tests/tests_config/test_validation.py

```diff
@@ -155,6 +155,39 @@ def test_invalid_run_parameter():
         next_run(config, runs, validate=False)
 
 
+@pytest.mark.parametrize("search_type", ["bayes", "grid", "random"])
+def test_nested_run_parameter(search_type):
+    sweep_config = {
+        "method": search_type,
+        "metric": {"name": "loss", "goal": "minimize"},
+        "parameters": {
+            "foo": {"values": [1, 2]},
+            "a.b": {"value": 2, "nested": True},
+            "a.c": {"value": 3, "nested": True},
+            "a.d.e": {"value": 4, "nested": True},
+        },
+    }
+    desired_run_config = {
+        # User gets the original parameters as specified
+        "foo": {"value": 1},
+        "a.b": {"value": 2},
+        "a.c": {"value": 3},
+        "a.d.e": {"value": 4},
+        # User will aso get any nested parameters in un-nested form
+        # so they can retreive them via wandb.config['a']['d']['e']
+        # as requested in https://github.com/wandb/client/issues/982
+        # where values inside their nested wandb.config objects
+        # could not be sweeped over in a sweep config.
+        "a": {"value": {"b": 2, "c": 3, "d": {"e": 4}}},
```

Comment:
we should not be returning nested config parameters unnested and unnested ... it makes the config 2x larger unnecessarily, and users will never need the access pattern on L172-175. they only need what is on L181